File: src/main/java/cz/kamenitxan/jakon/utils/validation/Validator.java
package cz.kamenitxan.jakon.utils.validation;

/**
 * Created by TPa on 11.08.16.
 */
public interface Validator {
	boolean isValid(Object... params);

	default boolean allValid(Object... params) {
		if (params == null) return false;
		for (Object o : params) {
			if (!isValid(o)) return false;
		}
		return true;
	}
}

File: src/main/java/cz/kamenitxan/jakon/validation/Validator.scala
package cz.kamenitxan.jakon.validation

import java.lang.annotation.Annotation
import java.lang.reflect.Field

import scala.language.implicitConversions

trait Validator {
	def isValid(value: String, a: Annotation, field: Field, data: Map[Field, String]): Option[ValidationResult]

	implicit def result2Opt(validationResult: ValidationResult): Option[ValidationResult] = {
		validationResult.toOpt
	}
}